pkg/virt-launcher/virtwrap/converter/converter_test.go

```diff
@@ -1520,6 +1523,49 @@ var _ = Describe("Converter", func() {
 			Expect(domainSpec.Devices.Rng).ToNot(BeNil())
 		})
 
+		table.DescribeTable("Validate that QEMU SeaBios debug logs are ",
+			func(toDefineVerbosityEnvVariable bool, virtLauncherLogVerbosity int, shouldEnableDebugLogs bool) {
+
+				var err error
+				if toDefineVerbosityEnvVariable {
+					err = os.Setenv(services.ENV_VAR_VIRT_LAUNCHER_LOG_VERBOSITY, strconv.Itoa(virtLauncherLogVerbosity))
+					Expect(err).To(BeNil())
+				}
+
+				domain := api.Domain{}
+
+				err = Convert_v1_VirtualMachineInstance_To_api_Domain(vmi, &domain, c)
+				Expect(err).To(BeNil())
+
+				if domain.Spec.QEMUCmd == nil || (domain.Spec.QEMUCmd.QEMUArg == nil) {
+					return
+				}
+
+				if shouldEnableDebugLogs {
+					Expect(domain.Spec.QEMUCmd.QEMUArg).Should(ContainElements(
+						api.Arg{Value: "-chardev"},
+						api.Arg{Value: "file,id=firmwarelog,path=/tmp/qemu-firmware.log"},
+						api.Arg{Value: "-device"},
+						api.Arg{Value: "isa-debugcon,iobase=0x402,chardev=firmwarelog"},
+					))
+				} else {
+					Expect(domain.Spec.QEMUCmd.QEMUArg).ShouldNot(Or(
+						ContainElements(api.Arg{Value: "-chardev"}),
+						ContainElements(api.Arg{Value: "file,id=firmwarelog,path=/tmp/qemu-firmware.log"}),
+						ContainElements(api.Arg{Value: "-device"}),
+						ContainElements(api.Arg{Value: "isa-debugcon,iobase=0x402,chardev=firmwarelog"}),
+					))
+				}
+
+				err = os.Unsetenv(services.ENV_VAR_VIRT_LAUNCHER_LOG_VERBOSITY)
```

Comment:
I know this is an edge case, but it might make sense to leave the env var alone if `toDefineVerbosityEnvVariable` is `true`

Reply:
Makes sense  :ok_hand: